src/workers/transactions/userBoughtCores.ts

```diff
@@ -0,0 +1,54 @@
+import { CioTransactionalMessageTemplateId, sendEmail } from '../../common';
+import {
+  UserTransaction,
+  UserTransactionProcessor,
+  UserTransactionStatus,
+} from '../../entity/user/UserTransaction';
+import type { TypedWorker } from '../worker';
+
+const purchaseProcessors = [
+  // UserTransactionProcessor.AppleStoreKit,
+  UserTransactionProcessor.Paddle,
+];
+
+export const userBoughtCores: TypedWorker<'api.v1.user-transaction'> = {
+  subscription: 'api.user-bought-cores',
+  handler: async ({ data }, con, logger): Promise<void> => {
+    const transaction = await con.getRepository(UserTransaction).findOneBy({
+      id: data.transaction.id,
+    });
+
+    if (!transaction) {
+      logger.error(
+        { transactionId: data.transaction.id },
+        'Transaction not found',
+      );
+      return;
+    }
+
+    if (!purchaseProcessors.includes(transaction.processor)) {
+      return;
+    }
+
+    if (transaction.status !== UserTransactionStatus.Success) {
+      return;
+    }
```

Comment:
One thing I am worried is that technically transaction could be updated again after success status and then we will send the email again?

Reply:
As a quick solution we could set a flag for this.

Reply:
Yeah, I think I should add it. Also need it for regular awards too

Reply:
But for regular awards it will be a notification so it will dedup from that?

Reply:
You're right, it dedupes, so no need for flag on awards

Reply:
Added now